lib/standards.h

```diff
@@ -93,10 +104,17 @@ struct Standards {
             cpp = CPP20;
             return true;
         }
+        if (str == "c++23" || str == "C++23") {
```

Comment:
could we lowercase str and reuse `getCpp` somehow? something like:

    stdValue = str;
    cpp = getCpp(lower(str));
    return cpp != CPPLatest || lower(str) == getCPP();

Reply:
Done - but with some slightly different logic